### ISSUE TITLE
TypeError: passes is not a function

### ISSUE THREAD

**Issue by jdcrecur** [mikeerickson/validatorjs#66]
I just found you're awesome JS Laravel validator! Nice work, was thinking about building one similar till i found this.

I am using this within AngularJS.

When i run this with errors in the form, the error cb is called fine (note there is no passes). However if I run this when the form passes the validators i get an error: "TypeError: passes is not a function" which is expected:

```
var validation = new Validator( dataToValidate, validationRules );
validation.fails(function(){
      Toolbox.metaFromFlatObject( $scope.stockTypeMeta, validation.errors.all(), 'error' );
});
```

Now, when i add the passes cb I get unexpected results.
Failing the form results in the fails cd being run but also an error in the console: "TypeError: fails is not a function"
Passing the form results in "TypeError: passes is not a function" but the cb is not called.

```
var validation = new Validator( dataToValidate, validationRules );
validation.fails(function(){
     Toolbox.metaFromFlatObject( $scope.stockTypeMeta, validation.errors.all(), 'error' );
});
validation.passes(function(){
     alert('passed!' );
});
```

If i chain the passes directly after the initialiser I can get the passes cb to run, however immediately after it has run i get an error "TypeError: Cannot read property 'fails' of undefined":

```
var validation = new Validator( dataToValidate, validationRules).passes(function(){
   alert('passed!' );
});

validation.fails(function(){
    Toolbox.metaFromFlatObject( $scope.stockTypeMeta, validation.errors.all(), 'error' );
});
```

Chaining the fails onto the end results in "TypeError: fails is not a function"

How can I get this working?

**Comment by jdcrecur** [comment]
OK, I kind of got it worked

A failure results in the errors being shown with no javascript errors in the console.

The  `if( validation.passes() ){`  runs when all validators passed, however it still throws an error into the console "TypeError: passes is not a function"

```
var validation = new Validator( dataToValidate, validationRules);
if( validation.passes() ){
    broadcastService.activityIndicator(true);
    stockTypesService.saveNew( dataToValidate, function( err, data ){
       if( err ) {
         Toolbox.metaFromFlatObject( $scope.stockTypeMeta, err.errors, 'error' );
       } else {
           $scope.getExisting();
       }
       broadcastService.activityIndicator(false);
   });
}
validation.fails(function(){
     Toolbox.metaFromFlatObject( $scope.stockTypeMeta, validation.errors.all(), 'error' );
});
```

**Comment by iamdtang** [comment]
Hey John,

Could you show me what your values are for `dataToValidate` and `validationRules`? Do you have asynchronous validation rules?

**Comment by jdcrecur** [comment]
console.log of dataToValidate : 
`Object {stock_type_code: "nrm", description: "Natural Resource - Mining"}`

console.log of the validationRules:
`Object {stock_type_code: "required|max:25", description: "required|max:100"}`

Then an error with the following trace route:

```
TypeError: passes is not a function
    at Object.resolvedAll [as onResolvedAll] (http://share-coral.app/js/all.min.js:1369:5)
    at Object.AsyncResolvers.fire (http://share-coral.app/js/all.min.js:468:9)
    at Validator.checkAsync (http://share-coral.app/js/all.min.js:1403:18)
    at Validator.fails (http://share-coral.app/js/all.min.js:1561:16)
```

However, this passes the validation tests and actually fires  the rel. code. Here is the full snippet:

```
        var dataToValidate = Toolbox.metaToFlatObject( $scope.stockTypeMeta, 'val'),
            validationRules = Toolbox.metaToFlatObject( $scope.stockTypeMeta, 'validators');

        console.log( dataToValidate );
        console.log( validationRules );

        var validation = new Validator( dataToValidate, validationRules);
        if( validation.passes() ){
            broadcastService.activityIndicator(true);
            stockTypesService.saveNew( dataToValidate, function( err, data ){
                if( err ) {
                    Toolbox.metaFromFlatObject( $scope.stockTypeMeta, err.errors, 'error' );
                } else {
                    Toolbox.metaSectionTo( $scope.stockTypeMeta, 'error', false );
                    Toolbox.metaSectionTo( $scope.stockTypeMeta, 'val', '' );
                    $scope.getExisting();
                }
                broadcastService.activityIndicator(false);
            });
        }
        validation.fails(function(){
            Toolbox.metaFromFlatObject( $scope.stockTypeMeta, validation.errors.all(), 'error' );
        });
```

**Comment by garygreen** [comment]
Fixed in 84624cb9bcea7b3fe18bba0fe077007624fd94a9

**Comment by jdcrecur** [comment]
Ah great! Is this new release in bower too?

**Comment by ssylvia** [comment]
I just downloaded the latest commit https://github.com/skaterdav85/validatorjs/commit/84624cb9bcea7b3fe18bba0fe077007624fd94a9 and it doesn't look like this is fixed yet. I have the following code:

```
const validator = new Validator(dataToValidate, validationRules);

if (validator.hasAsync) {
    validator.passes(() => {
        console.log('pass');
    });
   validator.fails(() => {
       console.log('fail');
   });
} else {
  console.log(validator.passes());
}
```

Anytime there is a async validation rule, I get the following error:
TypeError: fails is not a function(…) "TypeError: fails is not a function
    at Object.resolvedAll [as onResolvedAll]

or:
TypeError: passes is not a function(…) "TypeError: passes is not a function
    at Object.resolvedAll [as onResolvedAll]

**Comment by garygreen** [comment]
Are you running on node or browser? If browser did you grunt?

**Comment by iamdtang** [comment]
@johnc1984 Is this fixed? If not, try the latest version 2.0.7

**Comment by garygreen** [comment]
It's fixed.
